frame/vamm/src/lib.rs

```diff
@@ -793,7 +806,8 @@ pub mod pallet {
 			let mut vamm_state = Self::get_vamm_state(&config.vamm_id)?;
 
 			// Perform twap update before swapping assets.
-			Self::update_vamm_twap(config.vamm_id, &mut vamm_state, config.asset, &None)?;
+			// Ignore errors from this function while doing swaps.
+			let _ = Self::update_vamm_twap(config.vamm_id, &mut vamm_state, &None);
```

Comment:
I ignored the errors in this function because if we try to perform more than one swap in a block we would get the error `AssetTwapTimestampIsMoreRecent`. But looking more closely you are right, the function `Self::update_vamm_twap` is not `transactional` and errors could carry over (but as we have the order `sanity checks` > `computation` > `runtime update`, this would not be the case AFAIK). That's not ideal though and I'll address that in a later PR. 

We actually only ignore errors in this function because of the twap timestamp verification.